include/openthread-types.h

```diff
@@ -336,6 +341,30 @@ typedef enum
 } otDeviceRole;
 
 /**
+ * This structure represents the MAC layer counter.
+ */
+typedef struct otMacCounter
+{
+    uint32_t mTx;                  ///< The total number of transmissions.
+    uint32_t mTxData;              ///< The number of transmitted data.
+    uint32_t mTxDataPoll;          ///< The number of transmitted data poll.
+    uint32_t mTxBeacon;            ///< The number of transmitted beacon.
+    uint32_t mTxBeaconRequest;     ///< The number of transmitted beacon request.
+    uint32_t mTxRetry;             ///< The number of retransmission times.
+    uint32_t mTxErrAck;            ///< The number of unacked transmissions.
```

Comment:
@jwhui , I separate mTxData into 2 counters, one for the data transmission request ack, the other does not request ack, and rename mTxDataErrAck to mTxDataUnAcked. They may be clearer.
Then mTxDataAcked determines the number of acked transmission. And mTxDataUnAcked determines the number of unacked.
I still keep mTxRetry. It is one metric when comparing with Silabs thread stack.